payjoin/src/receive/v1/mod.rs

```diff
@@ -870,54 +886,48 @@ pub(crate) mod test {
         let empty_candidate_inputs: Vec<InputPair> = vec![];
         let result = wants_inputs.try_preserving_privacy(empty_candidate_inputs);
         match result {
+            Ok(_) => panic!("try_preserving_privacy should fail with empty candidate inputs"),
             Err(err) => {
-                let debug_str = format!("{:?}", err);
-                assert!(
-                    debug_str.contains("Empty"),
-                    "Error should indicate 'Empty' but was: {}",
-                    debug_str
-                );
+                assert_eq!(err, SelectionError::from(InternalSelectionError::Empty));
             }
-            Ok(_) => panic!("try_preserving_privacy should fail with empty candidate inputs"),
```

Comment:
This match is doing an assert_eq logic. I think something like
```rs
assert_eq!(
  result.unwrap_err()
  SelectionError::from(InternalSelectionError::Empty),
  "try_preserving_privacy should fail with empty candidate inputs"
);
```
would have a bit more straightforward syntax